### PR TITLE
Feat/65 events repo

### PR DESCRIPTION
**Pull request by AACioffi**
## What?

Added Firestore-backed `EventsRepositoryFirestore` implementation with comprehensive test coverage. Implements #65.

This PR introduces the data access layer for events, implementing the repository pattern with Firestore as the backend, along with a full test suite.

## Why?

The EventsRepositoryFirestore provides persistent storage for events using Firestore. This completes the data layer of our events feature and follows the established repository pattern used elsewhere in the app (e.g., ToDo feature), ensuring consistency in our architecture.

## How?

- **EventsRepositoryFirestore**: Firestore implementation of the EventsRepository interface, based on the existing ToDo repository structure for consistency
- **Repository pattern**: Handles CRUD operations and data mapping between Firestore documents and Event model objects
- **Test suite**: Comprehensive Firestore tests following the same structure as ToDo repository tests

## Testing?

Added integration tests using Firebase Emulator Suite covering:
- Basic CRUD operations (add, get, getAll, edit, delete)
- Error handling (e.g., NoSuchElementException for missing documents)
- Optional field handling
- UID generation

## Screenshots

image

The code achieved >80% line coverage, and has successfully passed CI.

## Anything Else?

This repository can now be integrated with the EventsViewModel (from #63) to provide end-to-end functionality for the events feature.

---

### CODE REVIEW

File: app/src/main/java/com/android/gatherly/model/event/EventsRepositoryFirestore.kt [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  /** Removes the given user ID from the participants list of the event with the given ID. */
+  override suspend fun removeParticipant(eventId: String, userId: String) {
+    collection.document(eventId).update("participants", FieldValue.arrayRemove(userId)).await()
+  }
+
+  /**
+   * Converts a Firestore [DocumentSnapshot] into an [Event] object.
+   *
+   * @param doc The Firestore document representing an [Event].
+   * @return The constructed [EVent], or `null` if required fields are missing.
```
Comment (coco-jrdn) [inline app/src/main/java/com/android/gatherly/model/event/EventsRepositoryFirestore.kt]:
```suggestion
   * @return The constructed [Event], or `null` if required fields are missing.
```

File: app/src/main/java/com/android/gatherly/model/event/EventsRepositoryFirestore.kt [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    // Should probably enforce that creatorId matches currentUserId
+    val ownedEvent = event.copy(creatorId = currentUserId())
+    collection.document(ownedEvent.id).set(eventToMap(ownedEvent)).await()
+  }
+
+  /** Edits the event with the given ID to have the new values. */
+  override suspend fun editEvent(eventId: String, newValue: Event) {
+    val existing = getEvent(eventId)
+    if (existing.creatorId != currentUserId()) {
+      throw SecurityException("Only the creator can edit this event")
```
Comment (coco-jrdn) [inline app/src/main/java/com/android/gatherly/model/event/EventsRepositoryFirestore.kt]: Nitpick : improve error by adding the eventId in the message for easier debugging

File: app/src/androidTest/java/com/android/gatherly/model/event/EventsRepositoryFirestoreTest.kt [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+      repository.deleteEvent(event1.id)
+      fail("Expected SecurityException to be thrown")
+    } catch (e: SecurityException) {
+      // Expected
+    }
+  }
+
+  /** Verifies that addParticipant successfully adds a user to the event's participant list. */
+  @Test
+  fun addParticipant_adds_user_to_list() = runTest {
```
Comment (coco-jrdn) [inline app/src/androidTest/java/com/android/gatherly/model/event/EventsRepositoryFirestoreTest.kt]: Add test to check if adding the same participant twice doesn’t duplicate
